manifest/discovery/discovery.go

```diff
@@ -0,0 +1,289 @@
+	path = strings.TrimPrefix(path, "/")
+	return fmt.Sprintf("https://raw.githubusercontent.com/%s/master/%s", d.repo.GetFullName(), path)
+}
+
+// newReq constructs a GET request with context.
+func newReq(ctx context.Context, url string) *http.Request {
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		// if it fails, it's a programmer's error
+		panic(err)
+	}
+	return req.WithContext(ctx)
+}
+
+// loadManifest reads manifest file from repository and decodes it into object.
+func (d *Driver) loadManifest(ctx context.Context) error {
+	req := newReq(ctx, d.repositoryFileURL(manifest.Filename))
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode == http.StatusNotFound {
+		// outdated driver
+		d.Name = d.Language
+		d.Status = manifest.Inactive
+		return nil
+	} else if resp.StatusCode/100 != 2 { // 2xx
+		return fmt.Errorf("status: %v", resp.Status)
+	}
+
+	lang := d.Language
+	if err := d.Manifest.Decode(resp.Body); err != nil {
+		return err
+	}
+	// override language ID from manifest (prevents copy-paste of manifests)
+	d.Language = lang
+	if d.Name == "" {
+		d.Name = d.Language
+	}
+	return nil
+}
+
+// reMaintainer is a regexp for one line of MAINTAINERS file (Github handle is optional):
+//
+//		John Doe <john@domain.com> (@john_at_github)
+var reMaintainer = regexp.MustCompile(`^([^<(]+)\s<([^>]+)>(\s\(@([^\s]+)\))?`)
+
+// loadMaintainers reads MAINTAINERS file from repository and decodes it into object.
+//
+// Each line in a file should follow the format defined by reMaintainer regexp.
+func (d *Driver) loadMaintainers(ctx context.Context) error {
+	req := newReq(ctx, d.repositoryFileURL("MAINTAINERS"))
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode == http.StatusNotFound {
+		return nil
+	} else if resp.StatusCode/100 != 2 {
+		return fmt.Errorf("status: %v", resp.Status)
+	}
+
+	var out []Maintainer
+	sc := bufio.NewScanner(resp.Body)
+	for sc.Scan() {
+		line := strings.TrimSpace(sc.Text())
+		sub := reMaintainer.FindStringSubmatch(line)
+		if len(sub) == 0 {
+			continue
+		}
+		m := Maintainer{Name: sub[1], Email: sub[2]}
+		if len(sub) >= 5 {
+			m.Github = sub[4]
+		}
+		out = append(out, m)
+	}
+	d.Maintainers = out
+	return nil
+}
+
+type Options struct {
+	Organization  string // Github organization name
+	NamesOnly     bool   // driver manifest will only have Language field populated
+	NoMaintainers bool   // do not load maintainers list
+}
+
+// OfficialDrivers lists all available language drivers for Babelfish.
+func OfficialDrivers(ctx context.Context, opt *Options) ([]Driver, error) {
+	if opt == nil {
+		opt = &Options{}
+	}
+	if opt.Organization == "" {
+		opt.Organization = GithubOrg
+	}
+	cli := github.NewClient(nil)
+
+	var out []Driver
+	// list all repositories in organization
+	for page := 1; ; page++ {
```

Comment:
Can move this loop/functionality (listing all the repositories and getting the drivers). to another function.

Reply:
Done